README.md

```diff
@@ -361,6 +361,76 @@ secrets:
       name: xwiki-db-root-password
 ```
 
+## Configuring clustering
+
+Read about setting communication channels [here](https://www.xwiki.org/xwiki/bin/view/Documentation/AdminGuide/Clustering/).
+
+#### Docker Compose example
+
+```yaml
+version: '2'
+networks:
+  bridge:
+    driver: bridge
+services:
+  web:
+    build: .
+    container_name: xwiki-postgres-tomcat-web
+    depends_on:
+      - db
+    ports:
+      - "8080:8080"
+    environment:
+      - XWIKI_VERSION=${XWIKI_VERSION}
+      - DB_USER=${DB_USER}
+      - DB_PASSWORD=${DB_PASSWORD}
+      - DB_DATABASE=${DB_DATABASE}
+      - DB_HOST=xwiki-postgres-db
+      - CLUSTER=true
+      - CLUSTER_CHANNEL=udp
```

Comment:
I'm wondering if we shouldn't prefix all XWiki-related env properties with "XWIKI", WDYT?

Reply:
does XWIKI-related include variable for DB and JGroups ?
like CLUSTER above?

Reply:
CLUSTER and CLUSTER_CHANNEL  are XWiki properties, not jgroups one. So I would prefix with XWiki.

Now just checked and I see that we recently introduced INDEX_HOST and INDEX_PORT for solr, see
https://github.com/xwiki-contrib/docker-xwiki/blob/4ebe570a79aeed8eabf7a8b6b834a179efc4b8b1/template/xwiki/docker-entrypoint.sh#L126 and they were not prefixed with XWiki.

New idea: why not use the same property name than in the xwiki config file? That could make it simpler maybe. The properties are:
- `observation.remote.enabled`
- `observation.remote.channels`

This would mean also deprecating INDEX_HOST and INDEX_PORT and instead introduce a `solr.remote.url` property (see https://github.com/xwiki-contrib/docker-xwiki/blob/4ebe570a79aeed8eabf7a8b6b834a179efc4b8b1/template/xwiki/docker-entrypoint.sh#L149).

In this manner, we would have a good strategy for any future properties that we need to add.

WDYT?

Also, we could it more generically so that the user would be able to pass any property from xwiki.cfg or xwiki.properties using `docker run -e <property name>` and the entry point script would set them in xwiki.cfg or xwiki.properties. To know in which file to add, we could search the file to see if it contains the property name.

WDYT?

Reply:
Using the same property name seems good.

I think for more generic property user should bind-mount it's property.

Reply:
> I think for more generic property user should bind-mount it's property.

Actually I forgot that we already expose the properties files in the permanent directory so that's taken care of already. Well, except for the same reason as you're doing this PR, i.e. to be able to control it from a compose file and avoid using 2 lines of `sed`....

So IMO we should still implement the generic mechanism.

WDYT?

Reply:
okay we can do that, but in bash env variable naming we cannot have dot(.).
So solr.remote.url cannot be set as a variable name in bash
https://unix.stackexchange.com/questions/93532/exporting-a-variable-with-dot-in-it

So for generic mechanisim, we need a naming convention like replacing dot with dash, but they have also issues if some properties contain - in it's name

Reply:
Indeed, not easy. So we should probaby drop my idea and continue with what you did, i.e. `CLUSTER` and `CLUSTER_CHANNEL`. And for other cases, then the caller will simply need to modify the xwiki configuration files from the permanent directory.

Thanks for the discussion :)